aiohttp/helpers.py

```diff
@@ -124,7 +124,7 @@ def iscoroutinefunction(func: Any) -> bool:
         return asyncio.iscoroutinefunction(func)
 
 
-json_re = re.compile(r"(?:application/|[\w.-]+/[\w.+-]+?\+)json", re.IGNORECASE)
+json_re = re.compile(r"(?:application/|[\w.-]+/[\w.+-]+?\+)json$", re.IGNORECASE)
```

Comment:
Maybe mark the start as well?
```suggestion
json_re = re.compile(r"^(?:application/|[\w.-]+/[\w.+-]+?\+)json$", re.IGNORECASE)
```

Reply:
This regex is used with `.match` and is not a `re.MULTILINE` one, so the start marker does not make a difference. The majority regexps in aiohttp that are used in this kind of setup don't have it either. IMO adding it would just be a potential source of confusion.

(And IMO the whole `re.match` method shouldn't even exist, _it_ is just a source of confusion and `re.search` could be used instead of it everywhere. Maybe some day it annoys me enough that I'll start the journey trying to get it deprecated and eventually removed from Python. </rant>)

Reply:
Good to me